service/src/worker.rs

```diff
@@ -77,13 +77,10 @@ where
 			// FIXME: Websocket connectionto a worker  should stay once etablished.
 			let client = WsClientBuilder::default().build(&url).await?;
 			let blocks = blocks_json.clone();
-			thread::spawn(move || {
-				if let Err(e) = futures::executor::block_on(
-					client.request::<Vec<u8>>("sidechain_importBlock", blocks.into()),
-				) {
-					error!("sidechain_importBlock failed: {:?}", e);
-				}
-			});
+			if let Err(e) = client.request::<Vec<u8>>("sidechain_importBlock", blocks.into()).await
+			{
+				error!("sidechain_importBlock failed: {:?}", e);
+			}
```

Comment:
There was a reason why you spawned a new thread here, no? Is that reason gone now?

Reply:
Original reason: We deadlocked here, because while we were still holding the enclave write lock, sent a message to ourself and tried to get a hold of the enclave write lock again.
So I had two ideas how to solve that:
1) create a new thread, so that we can return and release the lock
2) release the write lock before we gossip.

I turned both ideas into reality, as I thought both are good. But as it turned out, this one is not.

Reply:
Ah okay I see, thanks for explaining. Yes then it's probably more prudent to revert this to the way it was.

Reply:
Ah, if it is no longer necessary, even better! Seems we are at the limits of our thread usage. :)